src/commands/install.ts

```diff
@@ -190,17 +191,30 @@ function resolveWebDependency(dep: string): DependencyLoc {
       foundEntrypoint['./'] ||
       foundEntrypoint['.'];
   }
-  // If the package was a part of the explicit whitelist, fallback to it's main CJS entrypoint.
-  if (!foundEntrypoint) {
-    foundEntrypoint = depManifest.main || 'index.js';
-  }
   if (typeof foundEntrypoint !== 'string') {
     throw new Error(`"${dep}" has unexpected entrypoint: ${JSON.stringify(foundEntrypoint)}.`);
   }
-  return {
-    type: 'JS',
-    loc: require.resolve(path.join(depManifestLoc || '', '..', foundEntrypoint)),
-  };
+  // Sometimes packages don't give an entrypoint, assuming you'll fall back to "index.js".
+  const isImplicitEntrypoint = !foundEntrypoint;
+  if (isImplicitEntrypoint) {
+    foundEntrypoint = 'index.js';
+  }
+
+  try {
+    return {
+      type: 'JS',
+      loc: require.resolve(path.join(depManifestLoc || '', '..', foundEntrypoint)),
+    };
+  } catch (err) {
+    // Type only packages! Some packages are purely for TypeScript (ex: csstypes).
+    // If no JS entrypoint was given or found, but a TS "types"/"typings" entrypoint
+    // was given, assume a TS-types only package and ignore.
+    if (isImplicitEntrypoint && (depManifest.types || depManifest.typings)) {
+      return {type: 'IGNORE', loc: ''};
+    }
```

Comment:
Ah this is clever. I can see now we weren’t checking before if that worked.

I _think_ `require.resolve()` here would work without that `foundEntrypoint` at the end, but practically that shouldn’t make a difference